railties/test/generators/shared_generator_tests.rb

```diff
@@ -91,8 +104,10 @@ def test_template_is_executed_when_supplied_an_https_path
     template = %{ say "It works!" }
     template.instance_eval "def read; self; end" # Make the string respond to read
 
-    generator([destination_root], template: path).expects(:open).with(path, 'Accept' => 'application/x-thor-template').returns(template)
-    quietly { assert_match(/It works!/, capture(:stdout) { generator.invoke_all }) }
+    generator = generator([destination_root], template: path)
+    assert_called_with(generator, :open, [path, 'Accept' => 'application/x-thor-template'], returns: template) do
```

Comment:
Okay, got it. Minitest needs a third argument to stub (the arguments to yield to the `open` block). This fixes it:

```ruby
def test_template_is_executed_when_supplied_an_https_path
  path = "https://gist.github.com/josevalim/103208/raw/"
  template = %{ say "It works!" }
  template.instance_eval "def read; self; end" # Make the string respond to read

  check_open = -> *args do
    assert_equal [ path, 'Accept' => 'application/x-thor-template' ], args
    template
  end

  generator([destination_root], template: path).stub(:open, check_open, template) do
    quietly { assert_match(/It works!/, capture(:stdout) { generator.invoke_all }) }
  end
end
```

Big fan of `byebug` which helped me track this down by the way. Try putting `require 'byebug'; byebug` in there the next time you get stuck :smile:

(I had to remove the `quietly` and `capture` so `byebug` was in the process, but in reguler use it's smooth sailing :boat:)